onyo/lib/tests/test_git.py

```diff
@@ -0,0 +1,264 @@
+    file = Path('a/test/asset_for_test.0').resolve()
+
+    # make sure the cache GitRepo object contains the file path to the asset
+    assert file in new_git.files
+
+    # delete the file (with a non-onyo function to invalid the cache) and then
+    # verify that the file stays in the cache after the deletion
+    Path.unlink(file)
+    new_git.stage_and_commit(file, message="Save change for test")
+    assert file in new_git.files
+    assert not file.exists()
+
+    # test GitRepo.clear_caches() fixes the cache
+    new_git.clear_caches(files=True)
+    assert file not in new_git.files
+
+
+def test_GitRepo_is_clean_worktree(repo: OnyoRepo) -> None:
+    """
+    `GitRepo.is_clean_worktree()´ must return True when the worktree is clean,
+    and otherwise (i.e. for changed, staged, and unstracked files) return False.
+    """
+    new_git = GitRepo(repo.git.root)
+    test_file = repo.git.root / "test_file.txt"
+
+    # initially clean
+    assert new_git.is_clean_worktree()
+
+    # a file created but not added (i.e. untracked) must lead to return False
+    test_file.touch()
+    assert test_file in new_git.files_untracked
+    assert not new_git.is_clean_worktree()
+
+    # a file added but not commit-ed (i.e. staged) must lead to return False
+    new_git.add(test_file)
+    assert test_file in new_git.files_staged
+    assert not new_git.is_clean_worktree()
+
+    # when commit-ed, the function must return True again
+    new_git.commit(test_file, 'commit-ed!')
+    assert new_git.is_clean_worktree()
+
+    # a file modified but not commit-ed (i.e. changed) must lead to return False
+    test_file.open('w').write('Test: content')
+    new_git.add(test_file)
+    assert test_file in new_git.files_staged
+    assert not new_git.is_clean_worktree()
+
+    # when commit-ed, the function must return True again
+    new_git.commit(test_file, 'commit-ed again!')
+    assert new_git.is_clean_worktree()
+
+
+@pytest.mark.repo_files('existing/directory/test_file.txt')
+def test_GitRepo_is_git_path(repo: OnyoRepo) -> None:
+    """
+    `GitRepo.is_git_path()` needs to identify and return True for `.git/*`,
+    `.gitignore`, `.gitattributes`, `.gitmodules`, etc., and otherwise return
+    False.
+    """
+    new_git = GitRepo(repo.git.root)
+
+    # Test the examples listed above:
+    assert new_git.is_git_path(new_git.root / ".git")
+    assert new_git.is_git_path(new_git.root / ".git" / "HEAD")
+    assert new_git.is_git_path(new_git.root / ".git" / "info")
+    assert new_git.is_git_path(new_git.root / ".gitignore")
+    assert new_git.is_git_path(new_git.root / ".gitattributes")
+    assert new_git.is_git_path(new_git.root / ".gitmodules")
+
+    # Must return False
+    assert not new_git.is_git_path(new_git.root)
+    assert not new_git.is_git_path(new_git.root / ".onyo")
+    assert not new_git.is_git_path(new_git.root / "existing")
+    assert not new_git.is_git_path(new_git.root / "existing" / "directory")
```

Comment:
This doesn't add anything I can see to the previous assertion.
What's missing, however, are not existing paths.
What would also be missing are file/dir names that contain `.git` (but don't start with it) or just `git` (including the start, because it's only supposed to match `.git` as the start of the name.

```suggestion
    assert not new_git.is_git_path(new_git.root / "existing" / "git_no_.git")
```